libs/plugin/include/hpx/plugin/detail/dll_dlopen.hpp

```diff
@@ -37,44 +32,47 @@
 #include <limits.h>
 
 #if !defined(HPX_HAS_DLOPEN)
-#error "This file shouldn't be included directly, use the file hpx/util/plugin/dll.hpp only."
+#error                                                                         \
+    "This file shouldn't be included directly, use the file hpx/plugin/dll.hpp only."
 #endif
 
 #if !defined(_WIN32)
-typedef void* HMODULE;
+using HMODULE = void*;
 #else
-typedef struct HINSTANCE__* HMODULE;
+using HMODULE = struct HINSTANCE__*;
 #endif
 
 ///////////////////////////////////////////////////////////////////////////////
 #if !defined(RTLD_LOCAL)
-#define RTLD_LOCAL 0        // some systems do not have RTLD_LOCAL
+#define RTLD_LOCAL 0    // some systems do not have RTLD_LOCAL
 #endif
 #if !defined(RTLD_DEEPBIND)
-#define RTLD_DEEPBIND 0     // some systems do not have RTLD_DEEPBIND
+#define RTLD_DEEPBIND 0    // some systems do not have RTLD_DEEPBIND
 #endif
 
 ///////////////////////////////////////////////////////////////////////////////
-#define MyFreeLibrary(x)      dlclose (x)
-#define MyLoadLibrary(x) \
-      reinterpret_cast<HMODULE>(dlopen(x, RTLD_GLOBAL | RTLD_LAZY))
-#define MyGetProcAddress(x,y) dlsym   (x, y)
+#define MyFreeLibrary(x) dlclose(x)
+#define MyLoadLibrary(x)                                                       \
+    reinterpret_cast<HMODULE>(dlopen(x, RTLD_GLOBAL | RTLD_LAZY))
+#define MyGetProcAddress(x, y) dlsym(x, y)
 
 ///////////////////////////////////////////////////////////////////////////////
 namespace hpx { namespace util { namespace plugin {
 
-    namespace very_detail
-    {
+    namespace very_detail {
+
         template <typename TO, typename FROM>
         TO nasty_cast(FROM f)
         {
-            union {
-                FROM f; TO t;
+            union
+            {
+                FROM f;
+                TO t;
             } u;
             u.f = f;
             return u.t;
         }
-    }
+    }    // namespace very_detail
```

Comment:
Unrelated but this one should probably go to the same place the `nasty_cast` from the coroutines went (unless we really need it of course...).

Reply:
Ah, good question. I replaced it with a `reinterpret_cast`. Do you know if we really can't get away with one of the `std` casts here?